packages/react/runtime/src/lifecycle/patch/snapshotPatchApply.ts

```diff
@@ -129,10 +129,20 @@ export function snapshotPatchApply(snapshotPatch: SnapshotPatch): void {
   }
 }
 
+declare global {
+  const __webpack_require__: (<T>(id: string) => T) | undefined;
+}
+
 /**
  * Evaluates a string as code with ReactLynx runtime injected.
  * Used for HMR (Hot Module Replacement) to update snapshot definitions.
  */
 function evaluate<T>(code: string): T {
+  // This is a workaround for webpack when `output.iife` is `true`.
+  // We should fully support main-thread HMR and remove this `evaluate` function entirely.
+  // See https://github.com/lynx-family/lynx-stack/issues/983 for more details.
+  if (typeof __webpack_require__ !== 'undefined') {
+    return new Function('__webpack_require__', `return ${code}`)(__webpack_require__);
+  }
```

Comment:
Adding `__webpack_require__` here in runtime might be harder to maintain for other bundling tools.

Since we need to access the local variable `__webpack_require__ `, can we just use `eval` here? Since it is only for dev, it may not affect production (I am not sure if eval is not a good choice)